producer/writer/message_writer.go

```diff
@@ -324,8 +329,17 @@ func (w *messageWriterImpl) scanMessageQueue() {
 			// to avoid meaningless attempts, wait for next tick to retry.
 			break
 		}
+		if !fullScan && len(msgsToWrite) == 0 {
```

Comment:
New messages will be written out every newWritesScanInterval. Every fullScanInterval we will be writing both new messages and retries

Reply:
it does not wait for a full scan interval though, this is always triggered every `newWritesInterval`, only once in a while, it will decide to do a full scan. So new writes are always written every `newWritesInterval`

Reply:
Say you have 10 msgs came in, and this thread did a newWritesScan, then 1 msg came in, after `newWritesInterval` the newWritesScan will be writing the new msg out, because new messages are always written to the front of the list. There is a test case for that, can check that out if you have doubt